utility/ibm_dns_cleanup.py

```diff
@@ -0,0 +1,92 @@
+
+    Usage:
+        cleanup_ibm_dns_record.py --ibm-cred <cred-file>
+        cleanup_ibm_dns_record.py (-h | --help)
+
+    Options:
+        -h --help             Shows the command usage
+        --ibm-cred <file>     API Credential file to access ibm cloud.
+"""
+
+
+def run(args: Dict) -> int:
+    """
+    Using the provided credential file, this method removes the orphan DNS entries from ibm cloud.
+
+    Arguments:
+        args: Dict - containing the key/value pairs passed by the user
+
+    Returns:
+        0 on success or 1 for failures
+    """
+    ibm_cred_file = args["--ibm-cred"]
+
+    with open(ibm_cred_file, "r") as ibm_cred_stream:
+        yh = yaml.safe_load(ibm_cred_stream)
+        ibm_cred = yh["globals"]["ibm-credentials"]
+        ibmc_client = get_ibm_service(
+            access_key=ibm_cred["access-key"], service_url=ibm_cred["service-url"]
+        )
+        dns_client = get_dns_service(access_key=ibm_cred["access-key"])
+        dns_zone = dns_client.list_dnszones(instance_id)
+        if dns_zone.get_status_code() != 200:
+            print(f"Failed to get dns zone for given instance id: {instance_id}")
+            return 1
+        dns_zone_id = get_dns_zone_id(ibm_cred["zone_name"], dns_zone.get_result())
+        resource = dns_client.list_resource_records(
+            instance_id=instance_id, dnszone_id=dns_zone_id
+        )
+        if resource.get_status_code() != 200:
+            print(f"Failed to get dns records from zone: {ibm_cred['zone_name']}")
+            return 1
+        records = resource.get_result()
+        resp = ibmc_client.list_instances(vpc_name=ibm_cred["vpc_name"])
+        if resp.get_status_code() != 200:
+            print("Failed to retrieve instances")
+            return 1
+        response = resp.get_result()
+        ip_address = [
+            i["primary_network_interface"]["primary_ipv4_address"]
+            for i in response["instances"]
+        ]
+
+        for record in records["resource_records"]:
+            if record["type"] == "A" and record["rdata"]["ip"] not in ip_address:
+                if record.get("linked_ptr_record"):
+                    print(f"Deleting PTR record {record['linked_ptr_record']['name']}")
+                    dns_client.delete_resource_record(
+                        instance_id=instance_id,
+                        dnszone_id=dns_zone_id,
+                        record_id=record["linked_ptr_record"]["id"],
+                    )
+
+                print(f"Deleting Address record {record['name']}")
+                dns_client.delete_resource_record(
+                    instance_id=instance_id,
+                    dnszone_id=dns_zone_id,
+                    record_id=record["id"],
+                )
+
+        print("\nSuccessfully removed the orphan DNS record from IBM environment\n")
+    return 0
+
+
+if __name__ == "__main__":
+    arguments = docopt(doc)
+    rc = run(arguments)
```

Comment:
Let's have a `try...except` block around the run method. It will allows to fail gracefully.

Reply:
Made changes